tools/bigquery-data-consolidator/src/list_all_projects.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# List all projects, where we have access, using a Service account!
+
+# import libraries.
+from googleapiclient.discovery import build
+import google.auth
+import re  
+#from subprocess import call
+
+def list_projects(creds_file, org_id):
+    """
+    Lists all projects in an organization.
+    Requires the Organization ID as the parameter.
+    """
+    # call('gcloud projects list')
+
+    # loading credentials from the SA Key file. 
+    credentials, _ = google.auth.load_credentials_from_file(creds_file)
+
+    # calling V1 API for all details of resources other than Folders
+    rm_v1_client = build('cloudresourcemanager', 'v1', credentials=credentials, cache_discovery=False) 
+
+    # calling V2 specifically to get a list of Folders. 
+    rm_v2_client = build('cloudresourcemanager', 'v2', credentials=credentials, cache_discovery=False) 
+
+    # setting the Organization ID 
+    ORGANIZATION_ID = org_id
+
+    # creating empty list for final list of projects
+    final_list_of_projects = []
+
+    # List all the projects under the organization without/outside folders
+    filter='parent.type="organization" AND parent.id="{}"'.format(ORGANIZATION_ID)
+    projects_under_org = rm_v1_client.projects().list(filter=filter).execute()
+
+    # List project IDs for these projects
+    all_projects = [p['projectId'] for p in projects_under_org['projects']]
+
+    # Create list of all folders under the organization
+    parent="organizations/"+ORGANIZATION_ID
+    folders_under_org = rm_v2_client.folders().list(parent=parent).execute()
+
+    # Are there are actually folders under the org? 
+    # if not we can return the list of projects now. 
+    if not folders_under_org:
+        # filter the projects starting with sys-<long-number>. 
+        # these seems to be programmatically created hence can be ignored. 
+        matcher = re.compile('sys-[0-9]')
```

Comment:
you run the re.compile('sys-[0-9]') command twice. once here and once on line 95. Might be best to take it out of the if statement and put it above line 58 so you dont need to compile twice